### ISSUE TITLE
no-reserved-keys on object property assignment

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#2826]
Could we lint properties assigned to objects like so?

``` js
// Linted
var foo = {
    delete: 'bar',
};

// Not linted
foo.delete = 'bar';
```

Re. https://github.com/eslint/eslint/issues/1144

**Comment by nzakas** [comment]
I'm sorry, I'm not sure what you're asking. It would help if you described your request as "This is what happens right now, and this is what I would like to happen."

**Comment by OliverJAsh** [comment]
With `no-reserved-keys` set, this is what we lint right now:

``` js
var foo = {
    delete: 'bar',
};
```

However, we are not catching these cases:

``` js
foo.delete = 'bar';
```

**Comment by michaelficarra** [comment]
@OliverJAsh This already exists: http://eslint.org/docs/rules/dot-notation#options

**Comment by nzakas** [comment]
Closing per last comment
